api/tests.py

```diff
@@ -0,0 +1,193 @@
+from .test_helpers import SmallboardTestCase
+from rest_framework import status
+from puzzles.models import Puzzle, PuzzleTag
+
+if "unittest.util" in __import__("sys").modules:
+    # Show full diff in self.assertEqual.
+    __import__("sys").modules["unittest.util"]._MAX_LENGTH = 999999999
```

Comment:
This seems a little strange to include here. I think you can just add `-vv` when running `./manage.py test` to get more info including the full diff. I think it'd be better to increase the verbosity when running tests so it applies to all tests rather than making the change just for these api tests.

Reply:
Oh nice, I didn't know about -v. But I think that doesn't prevent the test output from being truncated:

```
======================================================================
FAIL: test_create_puzzle (api.tests.ApiTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/src/smallboard/api/tests.py", line 49, in test_create_puzzle
    self.assertEqual(
AssertionError: {'id'[137 chars]G', 'tags': [], 'guesses': [], 'metas': [], 'f[25 chars]alse} != {'id'[137 chars]G', 'guesses': [], 'metas': [], 'feeders': [],[13 chars]alse}
```

Since this runs during import I believe it applies to all tests as long as you're including `api/tests.py` but it wouldn't apply if you ran eg `test answers`, so I could move it somewhere else (maybe api/test_helpers.py?)

Reply:
What if you use two `v`s as in literally `-vv` for extra verbosity?

Reply:
```
usage: manage.py test [-h] [--noinput] [--failfast] [--testrunner TESTRUNNER] [-t TOP_LEVEL]
                      [-p PATTERN] [--keepdb] [-r] [--debug-mode] [-d] [--parallel [N]]
                      [--tag TAGS] [--exclude-tag EXCLUDE_TAGS] [--pdb] [-b]
                      [-k TEST_NAME_PATTERNS] [--version] [-v {0,1,2,3}]
                      [--settings SETTINGS] [--pythonpath PYTHONPATH] [--traceback]
                      [--no-color] [--force-color]
                      [test_label [test_label ...]]
manage.py test: error: argument -v/--verbosity: invalid int value: 'v'
```

The previous comment was with the most permissive verbosity level (3).

Reply:
I see... I am surprised at unittest's behavior.
I wonder if you can move this unittest configuration setup to the root directory and have it always be picked up?
If not, I would prefer not changing the MAX_LENGTH at all, since I feel this sort of configuration should be done globally (and put in a global place if so) rather than done in a specific test, even if it applies globally. But if you feel strongly for including it, I'm fine with keeping it, too.

Reply:
I'll just delete it, probably simplest that way.